src/Aspire.Hosting.Azure/AzureConstructResource.cs

```diff
@@ -55,8 +56,7 @@ public override BicepTemplateFile GetBicepTemplateFile(string? directory = null,
         var generationPath = Directory.CreateTempSubdirectory("aspire").FullName;
         var moduleSourcePath = Path.Combine(generationPath, "main.bicep");
 
-        var provisioningContext = GetProvisioningContext();
-        var plan = resourceModuleConstruct.Build(provisioningContext);
+        var plan = resourceModuleConstruct.Build(ProvisioningContext);
```

Comment:
You still need a fallback

Reply:
It allows `null`.

https://github.com/Azure/azure-sdk-for-net/blob/c8bdcfb11c5283c34da4693d5a700e565d80c384/sdk/provisioning/Azure.Provisioning/src/Infrastructure.cs#L193

It uses the `context ??= ProvisioningContext.Provider.GetProvisioningContext();`.

@tg-msft - is that going to change at all with the changes to remove the DefaultInfrastructure you were mentioning?